proto/regen/data/v1/query.proto

```diff
@@ -0,0 +1,132 @@
+syntax = "proto3";
+
+package regen.data.v1;
+
+import "google/api/annotations.proto";
+import "google/protobuf/timestamp.proto";
+import "cosmos/base/query/v1beta1/pagination.proto";
+import "regen/data/v1/types.proto";
+
+option go_package = "github.com/regen-network/regen-ledger/x/data";
+
+// Query is the regen.data.v1 Query service
+service Query {
+  // ByHash queries data based on its ContentHash.
+  rpc ByIRI (QueryByIRIRequest) returns (QueryByIRIResponse) {
+    option (google.api.http).get = "/regen/data/v1/by-iri/{iri}";
+  }
+
+  // BySigner queries data based on signers.
+  rpc BySigner (QueryBySignerRequest) returns (QueryBySignerResponse) {
+    option (google.api.http).get = "/regen/data/v1/by-signer/{signer}";
+  }
+
+  // Signers queries signers based on IRI.
+  rpc Signers(QuerySignersRequest) returns (QuerySignersResponse) {
+    option (google.api.http).get = "/regen/data/v1/signers/{iri}";
+  }
+
+  // Resolvers returns the registered resolver URLs for the data IRI.
+  rpc Resolvers(QueryResolversRequest) returns (QueryResolversResponse) {
+    option (google.api.http).get = "/regen/data/v1/resolvers/{iri}";
+  }
+
+  // ResolverInfo returns information about a resolver.
+  rpc ResolverInfo(QueryResolverInfoRequest) returns (QueryResolverInfoResponse) {
+    option (google.api.http).get = "/regen/data/v1/resolver/{url}";
```

Comment:
```suggestion
    option (google.api.http).get = "/regen/data/v1/resolver";
```

I think we cannot use URL as a path param.

Reply:
Good catch. We'll have to use `regen/data/v1/resolver/?url=http://test.com` when using the endpoint.